src/Types.mo

```diff
@@ -0,0 +1,94 @@
+import Prim "mo:⛔";
+
+module {
+  public type Blob = Prim.Types.Blob;
+  public type Bool = Prim.Types.Bool;
+  public type Char = Prim.Types.Char;
+  public type Error = Prim.Types.Error;
+  public type ErrorCode = Prim.ErrorCode;
+  public type Float = Prim.Types.Float;
+  public type Int = Prim.Types.Int;
+  public type Int8 = Prim.Types.Int8;
+  public type Int16 = Prim.Types.Int16;
+  public type Int32 = Prim.Types.Int32;
+  public type Int64 = Prim.Types.Int64;
+  public type Nat = Prim.Types.Nat;
+  public type Nat8 = Prim.Types.Nat8;
+  public type Nat16 = Prim.Types.Nat16;
+  public type Nat32 = Prim.Types.Nat32;
+  public type Nat64 = Prim.Types.Nat64;
+  public type Principal = Prim.Types.Principal;
+  public type Region = Prim.Types.Region;
+  public type Text = Prim.Types.Text;
+
+  public type Hash = Nat32;
+  public type Iter<T> = { next : () -> ?T };
+  public type Order = { #less; #equal; #greater };
+  public type Result<T, E> = { #ok : T; #err : E };
+  public type Pattern = {
+    #char : Char;
+    #text : Text;
+    #predicate : (Char -> Bool)
+  };
```

Comment:
i wonder it's still better to keep that in Text. Ditto for the other data structure ones unless really necessary.

Reply:
> Ditto for the other data structure ones unless really necessary.

I suppose we could move the other other data structure ones such as `Map.Node`, although in that case it's used in `Map`, so we would otherwise need to redefine it in both places. Makes it a somewhat tricky design choice. 

The vision for this module is that it would be possible to access all types defined in the base library, i.e. it might be odd to include every type except for a few select ones such as `Pattern`. I'd also be fine with removing it and then reintroducing later based on feedback.